packages/legend-data-cube/src/stores/core/DataCubeEngine.tsx

```diff
@@ -270,6 +270,11 @@ export abstract class DataCubeEngine {
     source: DataCubeSource,
   ): V1_AppliedFunction | undefined;
 
+  // ---------------------------------- FILE UPLOAD ------------------------------
+  async ingestFileData(csvString: string): Promise<DataCubeSource | undefined> {
```

Comment:
no need, this is particular to the `engine` of `Legend DataCube`, so keep this method over there. Remember we are very conservative regarding adding more functionalities to `DataCubeEngine`